precompiles/pallet-democracy/DemocracyInterface.sol

```diff
@@ -0,0 +1,194 @@
+     *
+     * @param prop_index index of the proposal you want to second
+     * @param seconds_upper_bound A number greater than or equal to the current number of seconds.
+     * This is necessary for calculating the weight of the call.
+     */
+    function second(uint256 prop_index, uint256 seconds_upper_bound) external;
+
+    //TODO should we have an alternative `simple_second` where the upper bound is read from storage?
+
+    /**
+     * Vote in a referendum.
+     *
+     * @param ref_index index of the referendum you want to vote in
+     * @param aye `true` is a vote to enact the proposal; `false` is a vote to keep the status quo.
+     * @param vote_amount The number of tokens you are willing to lock if you get your way
+     * @param conviction How strongly you want to vote. Higher conviction means longer lock time.
+     * This must be an interget in the range 0 to 6
+     *
+     * @dev This function only supposrts `Standard` votes where you either vote aye xor nay.
+     * It does not support `Split` votes where you vote on both sides. If such a need
+     * arises, we should add an additional function to this interface called `split_vote`.
+     */
+    function stardard_vote(
+        uint256 ref_index,
+        bool aye,
+        uint256 vote_amount,
+        uint256 conviction
+    ) external;
+
+    /** Remove a vote for a referendumx.
+     *
+     * @dev Locks get complex when votes are removed. See pallet-democracy's docs for details.
+     * @param ref_index The index of the referendum you are interested in
+     */
+    function remove_vote(uint256 ref_index) external;
+
+    /**
+     * Delegate voting power to another account.
+     *
+     * @dev The balance delegated is locked for as long as it is delegated, and thereafter for the
+     * time appropriate for the conviction's lock period.
+     * @param representative The account to whom the vote shall be delegated.
+     * @param conviction The conviction with which yo uare delegating. This conviction is used for
+     * _all_ delegated votes.
+     * @param amount The number of tokens whose voting power shall be delegated.
+     */
+    function delegate(
+        address representative,
+        uint256 conviction,
+        uint256 amount
+    ) external;
+
+    /**
+     * Undelegatehe voting power
+     *
+     * @dev Tokens may be unlocked once the lock period corresponding to the conviction with which
+     * the delegation was issued has elapsed.
+     */
+    function un_delegate() external;
+
+    /**
+     * Unlock tokens that have an expired lock.
+     *
+     * @param target The account whose tokens should be unlocked. This may be any account.
+     */
+    function unlock(address target) external;
+}
+
+// These are the selectors generated by remix following this advice
+// https://ethereum.stackexchange.com/a/73405/9963
+// Eventually we will probably want a better way of generating these and copying them to Rust
```

Comment:
We now have a Rust macro for that.
I think it is still a good idea to keep the computed by hand selectors here, and verify they are valid in the Rust tests.

Reply:
Added those tests in eb8c40a